src/Native/Unix/System.Native/pal_interfaceaddresses.cpp

```diff
@@ -44,6 +44,12 @@ extern "C" int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onI
     for (ifaddrs* current = headAddr; current != nullptr; current = current->ifa_next)
     {
         uint32_t interfaceIndex = if_nametoindex(current->ifa_name);
+        // ifa_name may be an aliased interface name.
+        // Use if_indextoname to map back to the true device name.
+        char actualName[IF_NAMESIZE];
+        char* result = if_indextoname(interfaceIndex, actualName);
+        assert(result == actualName && result != nullptr);
```

Comment:
I ended up doing what you suggested. If `if_indextoname` ends up failing, we exit out from the shim call early. Managed code retries up to 3 times but will otherwise throw a `NetworkInformationException` with the generic message `An error was encountered while querying information from the operating system.`

I had initially tried to handle a lot more weird error cases but I decided it wasn't valuable or that much more robust, and it complicated the code quite a bit. With this version, we're at least not ignoring the obvious errors.